### PR TITLE
cast string to list

### PR DESCRIPTION
**Pull request by AEsir777**
Change:
- driver newly upadated cast to string
- change old castToUINT32 to new method
- split header and cpp

Next PR:
- fix NULL value displayed as ,,
- currently spetial character can only be escaped as contained in quote " "
- space

### CODE REVIEW

**Comment by AEsir777** [comment]
| Dataset | New   | Old|
|---------| -------- | ------- |
| 20k INT64[]| 337.95  |  2058.97    |
| 100k INT64[] | 2322.9    | 9847.36200 |
| 10k INT64[][]   | 741.39  | FAIL |

File: dataset/load-from-test/bracket_fail.csv [hunk 1]
```diff
@@ -0,0 +1 @@
+"(()"
```
Comment (acquamarin) [inline dataset/load-from-test/bracket_fail.csv:1]: Add a newline

Comment (acquamarin) [inline dataset/load-from-test/bracket_fail.csv:1]: Can you move copy related tests to `test/test_files/copy`

File: src/function/cast_utils.cpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+void StringCastUtils::castStringToBool(const char* input, uint64_t len, bool& result) {
+    if (!tryCastToBool(input, len, result)) {
+        throw common::ConversionException(
+            "Cast failed. " + std::string{input, len} + " is not in BOOL range.");
+    }
+}
+
+} // namespace function
+} // namespace kuzu
```
Comment (acquamarin) [inline src/function/cast_utils.cpp:57]: newline

File: src/include/function/cast/cast_utils.h [hunk 3]
```diff
@@ -86,6 +45,13 @@ struct IntegerCastOperation {
     }
 };
 
+namespace StringCastUtils {
```
Comment (acquamarin) [inline src/include/function/cast/cast_utils.h:48]: I don't think we need to create a new namespace for StringCastUtils

File: src/processor/operator/persistent/reader/csv/driver.cpp [hunk 4]
```diff
@@ -19,6 +19,152 @@ namespace processor {
...
+};
+
+struct SplitStringListOperation {
+    SplitStringListOperation(uint64_t& offset, common::ValueVector* resultVector)
+        : offset(offset), resultVector(resultVector) {}
+
+    uint64_t& offset;
+    ValueVector* resultVector;
+
+    void HandleValue(const char* start, const char* end, const CSVReaderConfig& csvReaderConfig) {
```
Comment (acquamarin) [inline src/processor/operator/persistent/reader/csv/driver.cpp:89]: We use camel case for function names/variables

File: test/c_api/value_test.cpp [hunk 5]
```diff
@@ -888,7 +888,7 @@ TEST_F(CApiValueTest, NodeValToString) {
     ASSERT_STREQ(str,
         "{_ID: 1:0, _LABEL: organisation, ID: 1, name: ABFsUni, orgCode: 325, mark: 3.700000, "
         "score: -2, history: 10 years 5 months 13 hours 24 us, licenseValidInterval: 3 years "
-        "5 days, rating: 1.000000, state: {revenue: 138, location: ['toronto', 'montr,eal'], "
+        "5 days, rating: 1.000000, state: {revenue: 138, location: ['toronto','montr,eal'], "
```
Comment (acquamarin) [inline test/c_api/value_test.cpp:891]: Can you still allow white space between list element?

**Review by acquamarin — APPROVED** [review]

File: src/processor/operator/persistent/reader/csv/driver.cpp [hunk 6]
```diff
@@ -19,6 +19,157 @@ namespace processor {
 
 ParsingDriver::ParsingDriver(common::DataChunk& chunk) : chunk(chunk), rowEmpty(false) {}
 
+void copyStringToVector(ValueVector* vector, uint64_t rowToAdd, std::string_view strVal,
```
Comment (acquamarin) [inline src/processor/operator/persistent/reader/csv/driver.cpp:22]: I guess you can remove this function?

File: src/processor/operator/persistent/reader/csv/driver.cpp [hunk 7]
```diff
@@ -19,6 +19,157 @@ namespace processor {
...
+        }
+        input++;
+    }
+    return false; // no corresponding closing bracket
+}
+
+struct CountPartOperation {
+    uint64_t count = 0;
+
+    void HandleValue(const char* start, const char* end, const CSVReaderConfig& config) { count++; }
```
Comment (acquamarin) [inline src/processor/operator/persistent/reader/csv/driver.cpp:75]: Use camel case

**Review by acquamarin — APPROVED** [review]

File: src/processor/operator/persistent/reader/csv/driver.cpp (hunk 6 above)
Comment (AEsir777) [reply src/processor/operator/persistent/reader/csv/driver.cpp:22]: cast to struct is still using this function

**Pull request merged**